src/Ucommerce.Transactions.Payments.Freepay/FreepayPaymentMethodService.cs

```diff
@@ -0,0 +1,397 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Web;
+using System.Xml.Linq;
+using Ucommerce.EntitiesV2;
+using Ucommerce.Extensions;
+using Ucommerce.Infrastructure;
+using Ucommerce.Infrastructure.Environment;
+using Ucommerce.Transactions.Payments.Common;
+using Ucommerce.Transactions.Payments;
+using Ucommerce.Web;
+using Ucommerce.Infrastructure.Logging;
+
+namespace Ucommerce.Transactions.Payments.Freepay
+{
+    /// <summary>
+    /// Freepay integration via hosted payment form.
+    /// </summary>
+    public class FreepayPaymentMethodService : ExternalPaymentMethodService
+    {
+        private readonly IAbsoluteUrlService _absoluteUrlService;
+        private readonly ICallbackUrl _callbackUrl;
+        private readonly IWebRuntimeInspector _webRuntimeInspector;
+        private readonly ILoggingService _loggingService;
+        private AbstractPageBuilder PageBuilder { get; set; }
+
+	    private const string API_ENDPOINT_URL = "https://mw.freepay.dk";
+
+	    /// <summary>
+        /// Initializes a new instance of the <see cref="FreepayPaymentMethodService"/> class.
+        /// </summary>
+        public FreepayPaymentMethodService(IWebRuntimeInspector webRuntimeInspector, IAbsoluteUrlService absoluteUrlService, ICallbackUrl callbackUrl, ILoggingService loggingService)
+		{
+		    _webRuntimeInspector = webRuntimeInspector;
+            _absoluteUrlService = absoluteUrlService;
+            _callbackUrl = callbackUrl;
+            _loggingService = loggingService;
+        }
+
+        /// <summary>
+        /// Renders the forms to be submitted to the payment provider.
+        /// </summary>
+        /// <param name="paymentRequest">The payment request.</param>
+        /// <returns>A string containing the html form.</returns>
+        public override string RenderPage(PaymentRequest paymentRequest)
+        {
+            throw new NotSupportedException("Freepay does not need a local form. Use RequestPayment instead.");
+        }
+
+        public override Payment RequestPayment(PaymentRequest paymentRequest)
+        {
+            if (paymentRequest.Payment == null)
```

Comment:
There is still a tiny possibility that payment request could be null, so adding:
```suggestion
            if (paymentRequest?.Payment == null)
```
Will help :)